File: src/bin/day01.rs
use std::collections::HashSet;
use std::fs;

fn solve(nums: &[i32]) -> Option<(i32, i32, i32)> {
    let hm: HashSet<i32> = nums.iter().cloned().collect();
    for a in nums {
        for b in nums {
            let c = 2020 - a - b;
            if hm.get(&c).is_some() {
                return Some((*a, *b, c));
            }
        }
    }
    None
}

fn main() {
    let contents = fs::read_to_string("inputs/day01.txt").expect("Something went wrong reading the file");
    let nums: Vec<i32> = contents
        .lines()
        .map(|s| s.parse::<i32>().unwrap())
        .collect();

    if let Some((a, b, c)) = solve(&nums) {
        println!("{} * {} * {} = {}", a, b, c, a * b * c);
    } else {
        println!("could not find solution");
    }
}

File: julia/day01.jl
input = open("../inputs/day01.txt") do f
    read(f, String)
end

function parseInput(inp::String)::Array{Int64,1}
    [parse(Int64, ln) for ln in split(inp, "\n")]
end

function main()
    set = parseInput(input)
    for a in set
        for b in set
            c = 2020 - a - b
            if c in set
                println("$a + $b + $c = 2020")
                println("$a * $b * $c = $(a * b * c)")
                return
            end
        end
    end
end

main()